features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/ParserStageSequenceBuilder.java

```diff
@@ -715,6 +822,46 @@ public boolean equals(Object o) {
 		}
 	}
 
+	/**
+	 * Match a string terminated by a character in a list of end tokens containing only characters matching the given
+	 * predicate.
+	 */
+	static abstract class MatchOnlyUntil<R> extends MatchUntil<R> {
+		private final CharPredicate m_charMatcher;
+
+		MatchOnlyUntil(BiConsumer<ParserState,R> consumer, String end, CharPredicate charMatcher) {
+			super(consumer, end);
+			m_charMatcher = charMatcher;
+		}
+
+		@Override
+		public AcceptResult acceptChar(ParserStageState state, char c) {
+			for (char end : getEnd()) {
+				if (end == c || ! m_charMatcher.test(c)) {
+					return AcceptResult.COMPLETE_WITHOUT_CONSUMING;
+				}
+			}
+			// TODO: Make this more efficient?
```

Comment:
Is there a way to make this more efficient? `c` is a single character, so comparing it against a set of known white-space characters (which would be truncated by `trim()`) may be faster.